geth/params/config_test.go

```diff
@@ -215,17 +233,43 @@ var loadConfigTestCases = []struct {
 		},
 	},
 	{
-		`default boot cluster (Ropsten Dev)`,
+		`default boot nodes (Ropsten Dev)`,
 		`{
 			"NetworkId": 3,
 			"DataDir": "$TMPDIR"
 		}`,
 		func(t *testing.T, dataDir string, nodeConfig *params.NodeConfig, err error) {
 			require.NoError(t, err)
-			require.True(t, nodeConfig.BootClusterConfig.Enabled, "boot cluster is expected to be enabled by default")
+			require.True(t, nodeConfig.BootClusterConfig.Enabled, "boot nodes are expected to be enabled by default")
+
+			enodes := nodeConfig.BootClusterConfig.BootNodes
+			require.Len(t, enodes, 2)
+		},
+	},
+	{
+		`illegal cluster config file`,
+		`{
+			"NetworkId": 3,
+			"DataDir": "$TMPDIR",
+			"ClusterConfigFile": "/file/does/not.exist"
+		}`,
+		func(t *testing.T, dataDir string, nodeConfig *params.NodeConfig, err error) {
+			require.Error(t, err, "error is expected, not thrown")
+		},
+	},
+	{
+		`valid cluster config file`,
+		`{
+			"NetworkId": 3,
+			"DataDir": "$TMPDIR",
+			"ClusterConfigFile": "$TMPDIR/cluster.json"
+		}`,
+		func(t *testing.T, dataDir string, nodeConfig *params.NodeConfig, err error) {
+			require.NoError(t, err)
+			require.True(t, nodeConfig.BootClusterConfig.Enabled, "bootnodes are expected to be enabled after loading file")
 
 			enodes := nodeConfig.BootClusterConfig.BootNodes
-			require.Len(t, enodes, 4)
+			require.True(t, len(enodes) == 2)
```

Comment:
Shouldn't this have caused the CI build to fail before this fix?

Reply:
Test here uses the simulated cluster config file defined as constant above. No semantic check of a minimum number. Do we validate the number of configured nodes somewhere?

Reply:
Using `develop` here, if I change `4` to `5` I see this:

```
        config_test.go:374: test: default boot cluster (Ropsten Dev)
        config_test.go:374: test: default boot cluster (Ropsten Prod)
        Error Trace:    config_test.go:243
                        config_test.go:377
        Error:          "[enode://fa63a6cc730468c5456eab365b2a7a68a166845423c8c9acc363e5f8c4699ff6d954e7ec58f13ae49568600cff9899561b54f6fc2b9923136cd7104911f31cce@163.172.168.202:30303 enode://90cbf961c87eb837adc1300a0a6722a57134d843f0028a976d35dff387f101a2754842b6b694e50a01093808f304440d4d968bcbc599259e895ff26e5a1a17cf@51.15.194.39:30303 enode://dffef3874011709b12d1e540d83ddb19a9db8614ad9151d05bcf813585e45cbebba5aaea223fe315786c401d8cecb1ad2de9f179680c536ea30311fb21fa934b@188.166.100.178:30303 enode://03f3661686d30509d621dbe5ee2e3082923f25e94fd41a2dd8dd34bb12a0c4e8fbde52247c6c55e86dc209a8e7c4a5ae56058c65f7b01734d3ab73818b44e2a3@188.166.33.47:30303]" should have 5 item(s), but has 4
```

those are real node addresses. Wonder where they're coming from...

Reply:
Will investigate, also seen some references to terminus static peers I though I renamed. But diff shows wrong renaming. 🤔

Reply:
oh, it's a static asset, nvm!